src/games/strategy/triplea/ui/export/ScreenshotExporter.java

```diff
@@ -0,0 +1,205 @@
+package games.strategy.triplea.ui.export;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics2D;
+import java.awt.geom.AffineTransform;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import javax.imageio.ImageIO;
+import javax.swing.JFileChooser;
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
+import javax.swing.filechooser.FileFilter;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.framework.ui.SaveGameFileChooser;
+import games.strategy.engine.history.HistoryNode;
+import games.strategy.engine.history.Round;
+import games.strategy.triplea.ui.IUIContext;
+import games.strategy.triplea.ui.MapPanel;
+import games.strategy.triplea.ui.TripleAFrame;
+import games.strategy.triplea.ui.mapdata.MapData;
+import games.strategy.ui.SwingComponents;
+import games.strategy.ui.Util;
+
+/**
+ * Exports a game screenshot.
+ */
+public final class ScreenshotExporter {
+  private final TripleAFrame frame;
```

Comment:
The `TripleAFrame` instance is used during the screenshot export for the following reasons:

1. To provide a parent component for various dialogs that are displayed.
1. To provide the `IUIContext` instance used to retrieve various screenshot properties (e.g. scale, title, color, etc.).
1. To provide the `MapPanel` instance used to draw the actual screenshot image.

(1) is easy to address by simply using a `null` parent component for all dialogs, which is what several methods in `SwingComponents` already do.  (Note that the TripleA code I've examined seems to inconsistently choose between displaying a dialog relative to an actual parent component or relative to the screen.  I've been choosing the former in this PR.)

(2) and (3) might require some thought.  One could simply inline the `IUIContext` and `MapPanel` as parameters, but I'm not sure if that's any better.  If not for (1), I probably would say inlining them is acceptable.  Otherwise, it seems that the `TripleAFrame` is simply being used as a sort of ad-hoc Parameter Object.